### ISSUE TITLE
Add "error mode" to the sweep options

### ISSUE THREAD

**Issue by mithro** [verilog-to-routing/vtr-verilog-to-routing#320]
vpr has the following options;
```
netlist options:
  --absorb_buffer_luts {on, off}
                    Controls whether LUTS programmed as buffers are absorbed by 
                    downstream logic (Default: on)
  --sweep_dangling_primary_ios {on, off}
                    Controls whether dangling primary inputs and outputs are 
                    removed from the netlist (Default: on)
  --sweep_dangling_nets {on, off}
                    Controls whether dangling nets are removed from the netlist (Default: on)
  --sweep_dangling_blocks {on, off}
                    Controls whether dangling blocks are removed from the netlist (Default: on)
  --sweep_constant_primary_outputs {on, off}
                    Controls whether primary outputs driven by constant values 
                    are removed from the netlist (Default: off)
```

It would be nice if the `--sweep` had a `error` mode where if there are any dangling `nets`|`blocks`|`primary_outputs` vpr would fail with an error. Kind of `-Werror` on C compilers....

In most cases for us, a dangling net means we have done something wrong.

Comes from https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/318 and converting the asserts warnings in https://github.com/verilog-to-routing/vtr-verilog-to-routing/commit/100c71ff8492d41c7f79e5e2de9fd3a10ec72e91.

I wonder if there should be a general infrastructure for making warnings into errors or similar?

**Comment by kmurray** [comment]
I agree a general infrastructure seems like the right approach. There are a variety of places and different end-user use-cases where different error/warning/info behaviour makes sense.

A related issue is coming up with a better logging infrastructure for VPR to control things like verbosity. Right now we do this pretty haphazardly with options like `--debug_clustering` and `--verbose_sweep`. to turn on more output in specific components of VPR.

Ideally there would be a single system which would allow us to control both error/warning levels, and verbosity on a per-module level.
